hack/components/bump-linux-bridge.sh

```diff
@@ -49,13 +56,65 @@ COPY --from=builder ${LINUX_BRIDGE_PATH}/bin/tuning ${LINUX_BRIDGE_TAR_CONTAINER
 RUN sha256sum ${LINUX_BRIDGE_TAR_CONTAINER_DIR}/bridge >${LINUX_BRIDGE_TAR_CONTAINER_DIR}/bridge.checksum
 RUN sha256sum ${LINUX_BRIDGE_TAR_CONTAINER_DIR}/tuning >${LINUX_BRIDGE_TAR_CONTAINER_DIR}/tuning.checksum
 EOF
-    ${OCI_BIN} build -t ${LINUX_BRIDGE_IMAGE_TAGGED} .
+}
+
+check_and_create_docker_builder() {
+    existing_builder=$(docker buildx ls | grep -w "$DOCKER_BUILDER" | awk '{print $1}' || true)
+    if [ -n "$existing_builder" ]; then
+        echo "Builder '$DOCKER_BUILDER' already exists. Using existing builder."
+        docker buildx use "$DOCKER_BUILDER"
+    else
+        echo "Creating a new Docker Buildx builder: $DOCKER_BUILDER"
+        docker buildx create --driver-opt network=host --use --name "$DOCKER_BUILDER"
+    fi
+}
+
+build_docker_image() {
+    docker buildx build --platform "${PLATFORMS}" --build-arg BUILD_ARCH="$ARCH" -t "${LINUX_BRIDGE_IMAGE_TAGGED}" . --push
+    docker buildx rm "$DOCKER_BUILDER"
+}
+
+build_podman_image() {
+    podman manifest rm "${LINUX_BRIDGE_IMAGE_TAGGED}" || true
+    podman rmi "${LINUX_BRIDGE_IMAGE_TAGGED}" || true
+    podman manifest create "${LINUX_BRIDGE_IMAGE_TAGGED}"
+
+    for platform in "${PLATFORM_LIST[@]}"; do
+        podman build --no-cache --build-arg BUILD_ARCH="$ARCH" --platform "$platform" --manifest "${LINUX_BRIDGE_IMAGE_TAGGED}" .
+    done
+}
+
+modify_dockerfile_for_platform_and_architecture() {
+    local dockerfile="$1"
+    # Modify Dockerfile to set platform and architecture
+    sed -i 's|^FROM registry.access.redhat.com/ubi8/ubi-minimal AS builder$|FROM --platform=linux/${BUILD_ARCH} registry.access.redhat.com/ubi8/ubi-minimal AS builder|' "$dockerfile"
+    sed -i 's|RUN GOFLAGS=-mod=vendor ./build_linux.sh|RUN GOFLAGS=-mod=vendor GOARCH=${TARGETARCH} GOOS=${TARGETOS} ./build_linux.sh|' "$dockerfile"
+    sed -i 's/^FROM registry.access.redhat.com\/ubi8\/ubi-minimal AS final$/FROM --platform=linux\/${TARGETARCH} registry.access.redhat.com\/ubi8\/ubi-minimal AS final/' "$dockerfile"
+}
```

Comment:
i dont understand why we sed -i instead directly building the dockerfile as needed please
after all the file is built in `create_dockerfile` right?
lets refactor this please

Reply:
We need to use `TARGETARCH` and `TARGETOS` in the Dockerfile, but these values are only available at runtime. Since the Dockerfile is generated using a `cat` command, directly inserting these variables during creation would result in empty values, leading to an incorrect Dockerfile. To address this, we generate the Dockerfile first and then update it using the `sed` command to properly set the values at runtime.

Reply:
Just out of curiosity please (no need to do it, because it means we need a new file)

Cant we do the same you did before, have those in a new actual Dockerfile (not one that is catted),
and supply them externally, exactly as in other places ?
```
ARG TARGETOS
ARG TARGETARCH
```

Reply:
I am a bit confused why it is different than all the other cases where we have a real Dockerfile (not catted)
with those as args, but we can take it offline, and unblock this PR, (because in any case, not sure we want a standalone Dockerfile for the bump / publish)

Reply:
Fixed this, Removed sed commands and used backslash to avoid substitution while creating file for few specific lines with cat command.